ui/src/dashboards/containers/DashboardPage.tsx

```diff
@@ -421,15 +483,29 @@ class DashboardPage extends Component<Props, State> {
   private handleSaveEditedCell = async (
     newCell: DashboardsModels.Cell | NewDefaultCell
   ): Promise<void> => {
-    const {dashboard, handleHideCellEditorOverlay} = this.props
+    const {dashboard} = this.props
 
     if (this.isExistingCell(newCell)) {
       await this.props.updateDashboardCell(dashboard, newCell)
     } else if (!this.isExistingCell(newCell)) {
       this.props.addDashboardCellAsync(dashboard, newCell)
     }
 
-    handleHideCellEditorOverlay()
+    this.handleHideCellEditorOverlay()
+  }
+
+  private handleShowCellEditorOverlay = cell => {
+    const {handleShowCellEditorOverlay} = this.props
+    handleShowCellEditorOverlay(cell)
+    this.setState({showCellEditorOverlay: true})
+  }
+
+  private handleHideCellEditorOverlay = () => {
+    const {handleHideCellEditorOverlay} = this.props
+    this.setState({showCellEditorOverlay: false})
+    window.setTimeout(() => {
+      handleHideCellEditorOverlay()
+    }, 400)
```

Comment:
@ischolten bump

Reply:
yes i can put it in a constant. there is an animation that is triggered by setting showCellEDitorOverlay to false. That function needs to wait the amount of time it takes for that animation to complete until it can fire